src/mame/neogeo/neogeocd.cpp

```diff
@@ -436,29 +440,29 @@ uint8_t ngcd_state::transfer_r(offs_t offset)
 
 void ngcd_state::transfer_w(offs_t offset, uint8_t data)
 {
-	uint8_t byteValue = data;
-	uint32_t sekAddress = 0xe00000+ (offset);
+	uint32_t const seek_address = (0xe00000 + offset) ^ 1;
 
-	if (!m_transfer_write_enable) {
+	if (!m_transfer_write_enable)
+	{
 //      return;
 	}
-	int address;
 
-	sekAddress ^= 1;
-
-	switch (m_active_transfer_area) {
+	switch (m_active_transfer_area)
+	{
 		case 0:                         // Sprites
-			address = (m_sprite_transfer_bank + (sekAddress & 0x0FFFFF));
+		{
+			uint32_t const address = (m_sprite_transfer_bank + (seek_address & 0x0fffff));
 
 			// address is swizzled a bit due to out sprite decoding
-			if ((address&3)==0) m_sprite_ram[address] = byteValue;
-			if ((address&3)==1) m_sprite_ram[address^3] = byteValue;
-			if ((address&3)==2) m_sprite_ram[address^3] = byteValue;
-			if ((address&3)==3) m_sprite_ram[address] = byteValue;
+			if ((address & 3) == 0) m_sprite_ram[address] = data;
+			if ((address & 3) == 1) m_sprite_ram[address ^ 3] = data;
+			if ((address & 3) == 2) m_sprite_ram[address ^ 3] = data;
+			if ((address & 3) == 3) m_sprite_ram[address] = data;
```

Comment:
Would this be better as a `switch` statement?  There are only two possibilities.

You could also express it like:
```c++
			m_sprite_ram[address ^ ((BIT(address, 0) == BIT(address, 1)) ? 0 : 3)] = data;